compiler/src/dotty/tools/backend/jvm/BCodeSkelBuilder.scala

```diff
@@ -238,11 +238,16 @@ trait BCodeSkelBuilder extends BCodeHelpers {
 
       val ps = claszSymbol.info.parents
       val superClass: String = if (ps.isEmpty) ObjectReference.internalName else internalName(ps.head.widenDealias.typeSymbol)
-      val interfaceNames = classBTypeFromSymbol(claszSymbol).info.interfaces map {
+      val interfaceNames0 = classBTypeFromSymbol(claszSymbol).info.interfaces map {
         case classBType =>
           if (classBType.isNestedClass) { innerClassBufferASM += classBType }
           classBType.internalName
       }
+      val interfaceNames =
+        if (claszSymbol.is(ModuleClass) && !interfaceNames0.contains("java/io/Serializable"))
```

Comment:
I think we should be careful here because in SyntheticMembers the writeReplace method is only generated if an object is serializable https://github.com/lampepfl/dotty/blob/c5e1f0ceedabbbcbf70d84f30ce265dda9a4c22f/compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala#L369

Reply:
Oh yeah that needs to become unconditional (with a comment mentioning that all objects are serialized in the backend)